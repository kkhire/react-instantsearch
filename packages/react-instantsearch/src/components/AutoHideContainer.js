import PropTypes from 'prop-types';

const AutoHideContainer = ({ canRefine, children, autoHideContainer }) =>
  !autoHideContainer || canRefine ? children : null;

AutoHideContainer.propTypes = {
  canRefine: PropTypes.bool.isRequired,
  children: PropTypes.node.isRequired,
  autoHideContainer: PropTypes.bool,
};

AutoHideContainer.defaultProps = {
  autoHideContainer: false,
};

export default AutoHideContainer;
