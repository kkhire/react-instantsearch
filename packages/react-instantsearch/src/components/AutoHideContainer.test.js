import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AutoHideContainer from './AutoHideContainer';

Enzyme.configure({ adapter: new Adapter() });

describe('AutoHideContainer', () => {
  it('expect to render when canRefine is true & autoHideContainer is false', () => {
    const props = {
      canRefine: true,
      // use autoHideContainer defaultProps => false
    };

    const wrapper = shallow(
      <AutoHideContainer {...props}>
        <p>Hello from content</p>
      </AutoHideContainer>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('expect to render when canRefine is false & autoHideContainer is false', () => {
    const props = {
      canRefine: false,
      // use autoHideContainer defaultProps => false
    };

    const wrapper = shallow(
      <AutoHideContainer {...props}>
        <p>Hello from content</p>
      </AutoHideContainer>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('expect to render when canRefine is true & autoHideContainer is true', () => {
    const props = {
      canRefine: true,
      autoHideContainer: true,
    };

    const wrapper = shallow(
      <AutoHideContainer {...props}>
        <p>Hello from content</p>
      </AutoHideContainer>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('expect to not render when canRefine is false & autoHideContainer is true', () => {
    const props = {
      canRefine: false,
      autoHideContainer: true,
    };

    const wrapper = shallow(
      <AutoHideContainer {...props}>
        <p>Hello from content</p>
      </AutoHideContainer>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
