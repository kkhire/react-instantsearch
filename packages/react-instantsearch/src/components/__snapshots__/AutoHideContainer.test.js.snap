// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoHideContainer expect to not render when canRefine is false & autoHideContainer is true 1`] = `""`;

exports[`AutoHideContainer expect to render when canRefine is false & autoHideContainer is false 1`] = `
<p>
  Hello from content
</p>
`;

exports[`AutoHideContainer expect to render when canRefine is true & autoHideContainer is false 1`] = `
<p>
  Hello from content
</p>
`;

exports[`AutoHideContainer expect to render when canRefine is true & autoHideContainer is true 1`] = `
<p>
  Hello from content
</p>
`;
